src/es-module-shims.js

```diff
@@ -135,41 +139,42 @@ const initPromise = featureDetectionPromise.then(() => {
     }
   }
   baselinePassthrough = esmsInitOptions.polyfillEnable !== true && supportsDynamicImport && supportsImportMeta && supportsImportMaps && (!jsonModulesEnabled || supportsJsonAssertions) && (!cssModulesEnabled || supportsCssAssertions) && !importMapSrcOrLazy && !self.ESMS_DEBUG;
-  if (!supportsImportMaps) {
+  if (hasDocument && !supportsImportMaps) {
     const supports = HTMLScriptElement.supports || (type => type === 'classic' || type === 'module');
     HTMLScriptElement.supports = type => type === 'importmap' || supports(type);
   }
   if (shimMode || !baselinePassthrough) {
-    new MutationObserver(mutations => {
-      for (const mutation of mutations) {
-        if (mutation.type !== 'childList') continue;
-        for (const node of mutation.addedNodes) {
-          if (node.tagName === 'SCRIPT') {
-            if (node.type === (shimMode ? 'module-shim' : 'module'))
-              processScript(node);
-            if (node.type === (shimMode ? 'importmap-shim' : 'importmap'))
-              processImportMap(node);
+    if (hasDocument) {
```

Comment:
I'm not sure if they can be joined because at line 179 there is a `return lexer.init;` Maybe I'm wrong, but I think that line should be executed even if there is no `document` object (e.g. inside the worker  scope).

Reply:
We should probably just always return `lexer.init` actually, completely unguarded, there's no need to stop that since it's hardly blocking anything and always being done anyway.

Reply:
Yes, exactly!

Reply:
All right. Thanks!